### PR TITLE
fix: Fix clone page issue where JSObject are not showing up in destination page when application is connected to git

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> Cloning the page with JSObject ends up in branchName param not getting included in the defaultResource object, this PR fixes the mentioned scenario by including the branchName from the parent page.

### Steps to reproduce:
1. Connect app with git
2. Add JSObject to page and bind the sample data to widget
3. Clone the page
4. Cloned page JSObject does not have `defaultResources.branchName` param hence will not show up or throws the resource not found error if user tries to edit it

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> JUnit testcase
> Manual testing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by AnaghHegde** [comment]
@abhvsn Is this raised via todays hackathon event ?

**Comment by abhvsn** [comment]
> @abhvsn Is the PR raised via todays hackathon event ?

Yes we are fixing the resource not found error here

**Review by AnaghHegde — APPROVED** [review]

**Comment by AnaghHegde** [comment]
@abhvsn Can we add couple of more tests that specifically checks branchName param for JSObjects specifically? We can take this up in a new PR.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/PageServiceTest.java [hunk 1]
```diff
@@ -546,19 +576,44 @@ public void clonePage_whenPageCloned_defaultIdsRetained() {
...
+                    List<ActionCollection> collections = tuple.getT3();
+                    assertThat(collections.size()).isEqualTo(1);
+                    ActionCollection collection = collections.get(0);
+                    assertThat(collection.getPublishedCollection()).isNull();
+                    assertThat(collection.getUnpublishedCollection()).isNotNull();
+                    assertThat(collection.getUnpublishedCollection().getPageId()).isEqualTo(clonedPage.getId());
+                    DefaultResources collectionDefaultResource = collection.getDefaultResources();
+                    assertThat(collectionDefaultResource.getPageId()).isNull();
+                    assertThat(collectionDefaultResource.getApplicationId()).isEqualTo(collection.getApplicationId());
+                    assertThat(collectionDefaultResource.getBranchName()).isEqualTo(branchName);
```
Comment (abhvsn) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/PageServiceTest.java:608]: @AnaghHegde we are checking the branchname here as well. Are we expecting anything else?

**Pull request merged**